File: SoftUni Lessons/Python Development/Python Basics April 2019/Past Exams and Problems/7 - 9 and 10 March 2019/03_world_snooker_championship.py
type_final = input()
type_ticket = input()
count_tickets = int(input())
trophy_pic = input()

price = 0

if type_final == "Quarter final":
    if type_ticket == "Standard":
        price = 55.50 * count_tickets
    elif type_ticket == "Premium":
        price = 105.20 * count_tickets
    elif type_ticket == "VIP":
        price = 118.90 * count_tickets
elif type_final == "Semi final":
    if type_ticket == "Standard":
        price = 75.88 * count_tickets
    elif type_ticket == "Premium":
        price = 125.22 * count_tickets
    elif type_ticket == "VIP":
        price = 300.40 * count_tickets
elif type_final == "Final":
    if type_ticket == "Standard":
        price = 110.10 * count_tickets
    elif type_ticket == "Premium":
        price = 160.66 * count_tickets
    elif type_ticket == "VIP":
        price = 400 * count_tickets

if 2500 < price <= 4000:
    price -= price * 0.1
if price > 4000:
    price -= price * 0.25
    trophy_pic = "N"

if trophy_pic == "Y":
    price += 40 * count_tickets

print(f"{price:.2f}")

File: SoftUni Lessons/JavaScript Development/JavaScript Basics March 2020/21. МИНАЛИ ИЗПИТИ/03. Online Exam - 9 and 10 March 2019/03. World Snooker Championship.js
function worldSnookerChampionship(params) {
    let period = params.shift();
    let ticket = params.shift();
    let count = Number(params.shift());
    let picture = params.shift();

    let prices = {
        'Quarter final': { 'Standard': 55.50, 'Premium': 105.20, 'VIP': 118.90 },
        'Semi final': { 'Standard': 75.88, 'Premium': 125.22, 'VIP': 300.40 },
        'Final': { 'Standard': 110.10, 'Premium': 160.66, 'VIP': 400 }
    };

    let price = prices[period][ticket] * count;

    if (price > 4000) {
        price *= 0.75;
        picture = 'N';
    } else if (price > 2500) {
        price *= 0.9;
    }

    if (picture == 'Y') {
        price += 40 * count;
    } 
    
    console.log(price.toFixed(2));
}


worldSnookerChampionship(['Final', 'Premium', '25', 'Y']);
// Expected Output: 3012.38

worldSnookerChampionship(['Semi final', 'VIP', '9', 'Y']);
// Expected Output: 2793.24

worldSnookerChampionship(['Quarter final', 'Standard', '11', 'N']);
// Expected Output: 610.50
